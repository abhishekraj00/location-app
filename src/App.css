*{
    margin: 0;
    padding: 0;
    box-sizing: border-box;
    font-family: 'Segoe UI', Tahoma, Geneva, Verdana, sans-serif;
}

.App{
    display: flex;
    justify-content: center;
    align-items: center;
    flex-direction: column;
    gap: 1rem;
}

.leaflet-container {
    width: 100wh; 
    height: 100vh;
  }