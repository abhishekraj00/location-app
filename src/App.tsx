import "./App.css";
import Navbar from "./components/Navbar";
import { BrowserRouter, Route, Routes } from "react-router-dom";
import Home from "./pages/Home";
import Map from "./pages/Map";
import { createContext, useState } from "react";

interface LocationData {
  location_name: string;
  time: string;
}

interface HomeProp {
  locations: LocationData[] | [];
  setLocations: React.Dispatch<React.SetStateAction<LocationData[] | []>>;
}

export const LocationsContext = createContext<HomeProp>({
  locations: [],
  setLocations: () => null,
});

function App() {
  const [locations, setLocations] = useState<LocationData[] | []>([]);
  return (
    <LocationsContext.Provider value={{ locations, setLocations }}>
      <div className="App">
        <BrowserRouter>
          <Navbar />
          <Routes>
            <Route path="/" element={<Home />} />
            <Route path="/map" element={<Map />} />
          </Routes>
        </BrowserRouter>
      </div>
    </LocationsContext.Provider>
  );
}

export default App;
