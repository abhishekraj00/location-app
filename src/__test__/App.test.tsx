import { render, screen } from "@testing-library/react";
import App, { LocationsContext } from "../App";
import renderer from "react-test-renderer";

describe("App", () => {
  
  test("should render correctly", () => {
    const tree = renderer.create(<App />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders without crashing", () => {
    render(
      <LocationsContext.Provider
        value={{ locations: [], setLocations: jest.fn() }}
      >
        <App />
      </LocationsContext.Provider>
    );
    const homeLink = screen.getByText("Home");
    const mapLink = screen.getByText("Map");
    expect(homeLink).toBeInTheDocument();
    expect(mapLink).toBeInTheDocument();
  });
});
