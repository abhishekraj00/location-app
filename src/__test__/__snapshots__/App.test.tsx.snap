// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should render correctly 1`] = `
<div
  className="App"
>
  <nav
    className="navbar"
  >
    <a
      className="nav-link"
      data-testid="navigation-locations-tab"
      href="/"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="nav-link"
      data-testid="navigation-map-tab"
      href="/map"
      onClick={[Function]}
    >
      Map
    </a>
  </nav>
  <div
    data-testid="loading"
  >
    Loading...
  </div>
</div>
`;
